trustmanager/keyfilestore.go

```diff
@@ -54,31 +151,58 @@ func (s *KeyFileStore) Name() string {
 func (s *KeyFileStore) AddKey(name, role string, privKey data.PrivateKey) error {
 	s.Lock()
 	defer s.Unlock()
-	return addKey(s, s.Retriever, s.cachedKeys, name, role, privKey)
+	err := addKey(s, s.Retriever, s.cachedKeys, name, role, privKey)
+	if err != nil {
+		return err
+	}
+	s.keyInfoMap[privKey.ID()] = KeyInfo{Gun: getGunFromFullID(name), Role: role}
+	return nil
 }
 
 // GetKey returns the PrivateKey given a KeyID
 func (s *KeyFileStore) GetKey(name string) (data.PrivateKey, string, error) {
 	s.Lock()
 	defer s.Unlock()
+	// If this is a bare key ID without the gun, prepend the gun so the filestore lookup succeeds
+	if keyInfo, ok := s.keyInfoMap[name]; ok {
+		name = filepath.Join(keyInfo.Gun, name)
+	}
 	return getKey(s, s.Retriever, s.cachedKeys, name)
 }
 
 // ListKeys returns a list of unique PublicKeys present on the KeyFileStore.
-func (s *KeyFileStore) ListKeys() map[string]string {
-	return listKeys(s)
+func (s *KeyFileStore) ListKeys() map[string]KeyInfo {
+	return s.keyInfoMap
 }
 
 // RemoveKey removes the key from the keyfilestore
 func (s *KeyFileStore) RemoveKey(name string) error {
 	s.Lock()
 	defer s.Unlock()
-	return removeKey(s, s.cachedKeys, name)
+	// If this is a bare key ID without the gun, prepend the gun so the filestore lookup succeeds
+	if keyInfo, ok := s.keyInfoMap[name]; ok {
+		name = filepath.Join(keyInfo.Gun, name)
+	}
+	err := removeKey(s, s.cachedKeys, name)
+	if err != nil {
+		return err
+	}
+	// Remove this key from our keyInfo map if we removed from our filesystem
+	if _, ok := s.keyInfoMap[name]; ok {
+		delete(s.keyInfoMap, name)
+	} else {
+		// This might be of the form GUN/ID  - try to delete without the gun
+		delete(s.keyInfoMap, filepath.Base(name))
```

Comment:
`keyInfoMap` never includes the gun -- this was a defensive check in case the user passed a name of `gun/ID` to `RemoveKey()`, we're just deleting `ID` from the `keyInfoMap`.